gli/io.py

```diff
@@ -0,0 +1,99 @@
+import json
+from gli.utils import save_data
+import numpy as np
+import scipy.sparse as sp
+
+
+class Attributes(object):
+    def __init__(
+        self,
+        name,
+        description=None,
+        data=None,
+        data_type=None,
+        data_format=None
+    ):
+        self.name = name
+        self.description = description
+        self.data = data
+        self.key = None
+        print("data_type: ", data_type)
+        print(type(data))
+        if type is not None:
+            self.type = data_type
+        else:
+            self.type = type(data)
```

Comment:
type may be `int`, `float` or `str` for now. We should raise TypeError if `data` is not in these formats. We need something like this (generated by chatgpt):
```python
import numpy as np
from scipy.sparse import isspmatrix

def detect_array_type(array):
    if isspmatrix(array) or isinstance(array, np.ndarray):
        if array.size == 0:
            raise ValueError("The input array is empty.")
        
        # Check for the first non-null element's type
        for element in array.flat:
            if element is not None:
                element_type = type(element)
                break
                
        if issubclass(element_type, int):
            return "int"
        elif issubclass(element_type, float):
            return "float"
        elif issubclass(element_type, str):
            return "str"
        else:
            raise TypeError("The input array contains unsupported data types.")
    else:
        raise TypeError("The input array must be a scipy sparse array or numpy array.")

```
Not sure if it works but the overall idea is such